### PR TITLE
Fixed wrong file type when file extension is in upper case

### PR DESCRIPTION
**Pull request by KochynVolodymyr**
### Motivate of the pull request
When user add files we detect wrong file type in case file extension is in upper case.

### Which part of the code base require a change

- [x] Library: compiler
- [ ] Plug-in: <Specify the plugin name>
- [ ] Engine
- [ ] Documentation
- [x] Regression tests
- [ ] Continous Integration (CI) scripts

### Screenshots
image

### CODE REVIEW

File: src/Compiler/CompilerDefines.cpp [hunk 1]
```diff
@@ -140,3 +142,7 @@ bool FOEDAG::target_device(const QString &target) {
                            qPrintable(QString("target_device %1").arg(target)));
   return (res == TCL_OK);
 }
+
+bool FOEDAG::IsEqual(const QString &str, const QString &s) {
```
Comment (ravic-rs) [inline src/Compiler/CompilerDefines.cpp:146]: This sounds like a place for the utils class, no ?

Comment (KochynVolodymyr) [reply src/Compiler/CompilerDefines.cpp]: Done

Comment (ravic-rs) [reply src/Compiler/CompilerDefines.cpp]: Why not the string utils ? How are we planning to differentiate between QtUtils and StringUtils ?
https://github.com/os-fpga/FOEDAG/blob/main/src/Utils/StringUtils.cpp

Comment (KochynVolodymyr) [reply src/Compiler/CompilerDefines.cpp]: StringUtils related to std::string. If we have something that related to Qt it should go to QtUtils

**Pull request merged**